### ISSUE TITLE
How to get started with JEE?

### ISSUE THREAD

**Issue by stklik** [unige-pinfo-2018/Exercises#1]
TL;DR: read/watch http://www.adam-bien.com/roller/abien/entry/setting_up_java_ee_7

I received a long email from @kaikoveritch about Java EE7 and how to deal with it.
I will answer inline:

> We were told that our projects would use Java EE 7 and I am currently trying some setup to that end. There is however a dizzying amount of versions of Java and I must admit I am getting a bit lost, especially with Eclipse, which seems to have its own mysterious inner workings.

Yes. Java EE 7, it can be a bit difficult to wrap your head around it.
Java EE and Java are two different things.  
In order to use Java EE, you will have to have an Eclipse version + JDK installed (I suggest a recent version). You can then start developing.

> I have Eclipse EE and (by default) JavaSE-1.8 (jre/jdk). I tried switching directly to J2EE-7 (I think that's the abreviated name if I am not mistaken) but couldn't find the means to do it from Eclipse. My deduction was that I had in fact not installed Java EE at all and I tried to download it. I downloaded the third (and last) update of Java EE 7 from Oracle's website and got a "glassfish" directory which only seems to be a web development component (Eclipse will not let me select it). I did a lot of glorious googling on that matter, but could not find any definitive (or working) way of switching versions and at this point, I am basically lost.

Java EE is a set of APIs that provide additional capabilities to Java (and some "black magic").
In order to execute JEE code you will have to run your code within a Java EE 7 execution environment which is usually one of the following: 

 * Tomcat
 * Wildfly
 * Glassfish

You should therefore look into connecting either your server (let's say Wildfly for example) either inside Eclipse  (which makes it easy) or learn how to deploy to the Wildfly environment from outside Eclipse (a bit more complicated)

> Should I simply switch to JavaSE-1.7 ? Is there a setting I missed ?

Don't. (Please!). Use Java 8 (or Java 9).  
Install a recent version of Wildfly, Glassfish or Tomcat, make sure you can execute the code.
To create a project with Java EE Maven dependencies already included, use Adam Bien's JEE7 archetype (similarly to what we did in class with the `quickstart` archetype).
http://www.adam-bien.com/roller/abien/entry/setting_up_java_ee_7

A tutorial on how to link Wildfly with Eclipse: http://www.ejbtutorial.com/j2ee/getting-started-with-j2ee-installing-wildfly-on-eclipse  
The article is already from 2016, but I think the basic process is still the same.

**Comment by stklik** [comment]
@unige-pinfo-2018/pinfo1 @unige-pinfo-2018/pinfo2  @unige-pinfo-2018/pinfo3

**Comment by kaikoveritch** [comment]
Thank you for the clarification and pointers !

**Comment by stklik** [comment]
In Eclipse: You will have to add a remote catalogue for Maven to find the com.airhacks archetype.
To do so, you click `Configure` in the dialog offered by the `New Maven Project` and add
`http://repo.maven.apache.org/maven2`.
![image](https://user-images.githubusercontent.com/235894/36979210-124ab9f8-2087-11e8-8f1e-44fe73c1b79c.png)
